syntax = "proto3";
import "google/protobuf/empty.proto";

// client send it to clientProxy
message requestProxy{
  int32 cid = 1; // client id
  int32 qid = 2; //query id
  string db_name = 3; // database name
  string table_name =4;
  string column_name =5; // database column name
  string op = 6; // operation
}


// keyServer send it to clientProxy
message requestDecResult{
  int32 cid = 1; // client id
  int32 qid = 2; //query id
  bytes decResult = 3;
}

// clientProxy send it to client
message responseResult {
  bytes result = 1;
}

message flag{
  bool flag =1;
}

service ClientProxyService {
  // clientProxy provides the interface, client remotes the call
  rpc RequestProxy(requestProxy) returns(responseResult);
  // clientProxy provides the interface, keyServer remotes the call
  rpc RequestResult (requestDecResult) returns(google.protobuf.Empty);
}